File: Riot/Utils/DisappearingMessages/MXRoomState+Retention.m
/*
Copyright 2025 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only
Please see LICENSE in the repository root for full details.
*/

#import "MXRoomState+Retention.h"
#import <MatrixSDK/MXEvent.h>

@implementation MXRoomState (Retention)

- (NSNumber *)vc_maxLifetimeSeconds
{
    NSArray<MXEvent *> *events = [self stateEventsWithType:kMXEventTypeStringRoomRetention];
    MXEvent *event = events.lastObject;
    if (!event.content[@"max_lifetime"]) {
        return nil;
    }
    NSNumber *maxLifetimeMs = event.content[@"max_lifetime"];
    if (![maxLifetimeMs isKindOfClass:NSNumber.class] || maxLifetimeMs.longLongValue <= 0) {
        return nil;
    }
    long long seconds = maxLifetimeMs.longLongValue / 1000;
    return @(seconds);
}

@end

File: Riot/Utils/DisappearingMessages/MXRoomState+Retention.swift
/*
Copyright 2025 New Vector Ltd.

SPDX-License-Identifier: AGPL-3.0-only
Please see LICENSE in the repository root for full details.
*/

import Foundation
import MatrixSDK

extension MXRoomState {
    /// Max lifetime in seconds from m.room.retention (MSC1763), or nil if not set.
    func vc_retentionMaxLifetimeSeconds() -> Int? {
        vc_maxLifetimeSeconds()?.intValue
    }
}
